File: solvers.julia/wpnp_2x2sym/coefs_wpnp_2x2sym.jl
function coefs_wpnp_2x2sym(data::Vector{Matrix{Float64}})

    A = data[1]
    B = data[2]

    C = zeros(ComplexF64,4,16)

    C[[14,24,31,33]] .= 64*A[3]^2
    C[[3,5,38,48]] .= -32*A[1]^2+32*A[2]^2
    C[[12,17,26,43]] .= 32*A[1]^2+32*A[2]^2
    C[[56,59]] .= -8*A[3]*B[5]-8*im*A[3]*B[6]
    C[[50,61]] .= 8*im*A[3]*B[6]-8*A[3]*B[5]
    C[60] = -8*A[1]*B[1]+8*A[2]*B[4]-8*im*A[1]*B[2]-8*im*A[2]*B[3]
    C[49] = -8*A[1]*B[1]+8*A[2]*B[4]+8*im*A[1]*B[2]+8*im*A[2]*B[3]
    C[55] = 8*A[1]*B[1]+8*A[2]*B[4]+8*im*A[1]*B[2]-8*im*A[2]*B[3]
    C[62] = -8*im*A[1]*B[2]+8*im*A[2]*B[3]+8*A[1]*B[1]+8*A[2]*B[4]

    C = C./repeat(sqrt.(sum(C.*conj(C),dims=2)),1,size(C,2))

    U(w1,x1,y1,z1) = [x1*w1^2*y1,x1*y1^2*w1,x1*w1^2*z1,x1*y1*z1*w1,x1*z1^2*w1,y1*x1*w1,y1^2*x1,z1*x1*w1,z1*y1*x1,z1^2*x1,x1*y1,x1*z1,w1,y1,z1,1]

    return C,U

end

File: solvers.python/py_wpnp_2x2sym/coefs_wpnp_2x2sym.py
import numpy as np

def coefs_wpnp_2x2sym(data):
    A = data[0].T.flat
    B = data[1].T.flat

    C = np.zeros((4,16),dtype=np.complex_)
    C.flat[[8,19,39,53]] = 64*A[2]**2
    C.flat[[1,25,32,59]] = -32*A[0]**2+32*A[1]**2
    C.flat[[4,22,42,50]] = 32*A[0]**2+32*A[1]**2
    C.flat[[46,61]] = -8*A[2]*B[4]-8*1j*A[2]*B[5]
    C.flat[[15,28]] = 8*1j*A[2]*B[5]-8*A[2]*B[4]
    C.flat[62] = -8*A[0]*B[0]+8*A[1]*B[3]-8*1j*A[0]*B[1]-8*1j*A[1]*B[2]
    C.flat[12] = -8*A[0]*B[0]+8*A[1]*B[3]+8*1j*A[0]*B[1]+8*1j*A[1]*B[2]
    C.flat[45] = 8*A[0]*B[0]+8*A[1]*B[3]+8*1j*A[0]*B[1]-8*1j*A[1]*B[2]
    C.flat[31] = -8*1j*A[0]*B[1]+8*1j*A[1]*B[2]+8*A[0]*B[0]+8*A[1]*B[3]

    C = C/np.sqrt(np.sum(C*np.conj(C),axis=1,keepdims=True))

    def U(w1,x1,y1,z1):
        return np.array([x1*w1**2*y1,x1*y1**2*w1,x1*w1**2*z1,x1*y1*z1*w1,x1*z1**2*w1,y1*x1*w1,y1**2*x1,z1*x1*w1,z1*y1*x1,z1**2*x1,x1*y1,x1*z1,w1,y1,z1,1])

    return C,U